tests/specs/builds/typescript.ts

```diff
@@ -52,5 +53,120 @@ export default testSuite(({ describe }, nodePath: string) => {
 			const content = await fixture.readFile('dist/index.js', 'utf8');
 			expect(content).toBe('console.log(1);\n');
 		});
+
+		test('resolves bare module paths ', async () => {
+			await using fixture = await createFixture({
+				src: {
+					'index.ts': outdent`
+					import { qux } from 'nested/exportee.js';
+					import { quux } from 'nested/deep/exportee.js';
+					console.log(qux, quux);
+					export { qux, quux };`,
+					'importee.ts': outdent`export const foo = () => 'foo';`,
+					nested: {
+						'importee.ts': outdent`export const bar = () => 'bar';`,
+						'exportee.ts': outdent`
+						import { foo } from 'importee.js';
+						import { baz } from 'nested/deep/importee.js';
+						export const qux = foo() + baz();`,
+						deep: {
+							'importee.ts': outdent`export const baz = () => 'baz';`,
+							'exportee.ts': outdent`
+							import { foo } from 'importee.js';
+							import { bar } from 'nested/importee.js';
+							import { baz } from 'nested/deep/importee.js';
+							export const quux = foo() + bar() + baz();`
+						},
+					},
+				},
+				'package.json': createPackageJson({
+					type: 'module',
+					exports: {
+						'.': './dist/index.js',
+						'./nested': './dist/nested/exportee.js',
+						'./deep': './dist/nested/deep/exportee.js',
+					},
+				}),
+				'tsconfig.json': createTsconfigJson({
+					compilerOptions: {
+						baseUrl: './src',
+					},
+				})
+			});
+
+			const pkgrollProcess = await pkgroll(['--env.NODE_ENV=development'], {
+				cwd: fixture.path,
+				nodePath,
+			});
+
+			expect(pkgrollProcess.exitCode).toBe(0);
+			expect(pkgrollProcess.stderr).toBe('');
+
+			/** @todo expect */
```

Comment:
I think that the preceding expectations (no errors during pkgrollProcess) are sufficient. On the other hand I couldn't come up with a simple test due to the complexity of the fixtures. Maybe on could reduce the latter.